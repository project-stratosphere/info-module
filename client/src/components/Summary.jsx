import PropTypes from 'prop-types';
import React from 'react';
import {
  GuestsSVG,
  BedroomsSVG,
  BedsSVG,
  BathsSVG,
} from './SVG';
import {
  Wrapper,
  HomeType,
  HomeTitle,
  HomeLocation,
  Features,
  Feature,
  StyledImage,
  OwnerName,
  SVGContainer,
} from './styles/Summary.styles';

const Summary = props => (
  <Wrapper
    direction="row"
  >
    <Wrapper
      direction="column"
    >
      <HomeType>
        {props.home_type}
      </HomeType>
      <HomeTitle>
        {props.title}
      </HomeTitle>
      <HomeLocation>
        {props.location}
      </HomeLocation>
      <Features>
        <Feature>
          <SVGContainer>
            <GuestsSVG />
          </SVGContainer>
          {props.property_features[0]} Guests
        </Feature>
        <Feature>
          <SVGContainer>
            <BedroomsSVG />
          </SVGContainer>
          {props.property_features[1]} Bedrooms
        </Feature>
        <Feature>
          <SVGContainer>
            <BedsSVG />
          </SVGContainer>
          {props.property_features[2]} Beds
        </Feature>
        <Feature>
          <SVGContainer>
            <BathsSVG />
          </SVGContainer>
          {props.property_features[3]} Baths
        </Feature>
      </Features>
    </Wrapper>
    <Wrapper
      direction="column"
    >
      <StyledImage
        src={props.owner[1]}
        alt="avatar not found"
      />
      <OwnerName>
        {props.owner[0]}
      </OwnerName>
    </Wrapper>
  </Wrapper>
);

Summary.propTypes = {
  home_type: PropTypes.string,
  title: PropTypes.string,
  location: PropTypes.string,
};

Summary.defaultProps = {
  home_type: 'PropTypes.string',
  title: 'PropTypes.string',
  location: 'PropTypes.string',
};

export default Summary;
