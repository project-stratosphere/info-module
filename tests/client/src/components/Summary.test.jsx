import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Summary from '../../../../client/src/components/Summary';

Enzyme.configure({ adapter: new Adapter() });

const owner = {
  name: 'Mock Owner Name',
  avatar_url: 'MockAvatarURL',
};

const propertyFeatures = {
  guests: 1,
  bedrooms: 1,
  beds: 1,
  baths: 1,
};

describe('Summary', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Summary
      home_type="Mock Home Type"
      title="Mock Title"
      location="Mock Location"
      owner={owner}
      property_features={propertyFeatures}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
