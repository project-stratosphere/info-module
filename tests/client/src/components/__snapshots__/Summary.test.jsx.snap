// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should render correctly 1`] = `
<styled.div
  direction="row"
>
  <styled.div
    direction="column"
  >
    <styled.div>
      Mock Home Type
    </styled.div>
    <styled.div>
      Mock Title
    </styled.div>
    <styled.div>
      Mock Location
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <GuestsSVG />
        </styled.div>
        1
         Guests
      </styled.div>
      <styled.div>
        <styled.div>
          <BedroomsSVG />
        </styled.div>
        1
         Bedrooms
      </styled.div>
      <styled.div>
        <styled.div>
          <BedsSVG />
        </styled.div>
        1
         Beds
      </styled.div>
      <styled.div>
        <styled.div>
          <BathsSVG />
        </styled.div>
        1
         Baths
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    direction="column"
  >
    <styled.img
      alt="avatar not found"
      src="MockAvatarURL"
    />
    <styled.div>
      Mock Owner Name
    </styled.div>
  </styled.div>
</styled.div>
`;
